diff --git a/src/exe/CMakeLists.txt b/src/exe/CMakeLists.txt
index ce162e7..7f2f3af 100755
--- a/src/exe/CMakeLists.txt
+++ b/src/exe/CMakeLists.txt
@@ -36,4 +36,5 @@ if(IS_MSVC)
 endif()
 
 COLMAP_ADD_EXECUTABLE(colmap_exe colmap.cc)
-set_target_properties(colmap_exe PROPERTIES OUTPUT_NAME colmap)
+#https://github.com/MicrosoftDocs/cpp-docs/issues/741
+set_target_properties(colmap_exe PROPERTIES OUTPUT_NAME colmap_exe)
