diff --git a/CMakeLists.txt b/CMakeLists.txt
index b05097c..f506106 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,10 @@ set(COLMAP_VERSION_NUMBER "3600")
 # Include CMake dependencies
 ################################################################################
 
-set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR})
+if (WIN32)
+    add_definitions(/bigobj)
+endif()
 
 include(CheckCXXCompilerFlag)
 
@@ -98,11 +101,12 @@ find_package(Eigen3 REQUIRED)
 
 find_package(FreeImage REQUIRED)
 
-find_package(Glog REQUIRED)
+find_package(glog REQUIRED)
 
 find_package(OpenGL REQUIRED)
-find_package(Glew REQUIRED)
+find_package(GLEW REQUIRED)
 find_package(Git)
+find_package(flann)
 
 if(CGAL_ENABLED)
     set(CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE TRUE)
@@ -245,6 +249,10 @@ if(Qt5_POSITION_INDEPENDENT_CODE)
     if(CUDA_ENABLED AND NOT IS_MSVC)
         set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} --compiler-options -fPIC")
     endif()
+	# define only if cuda is not installed
+	if (NOT WIN32)
+		add_definitions(-fPIC)
+	endif()					
 endif()
 
 # Enable automatic compilation of Qt resource files.
@@ -276,18 +284,7 @@ set(COLMAP_LINK_DIRS
 
 set(COLMAP_EXTERNAL_LIBRARIES
     ${CMAKE_DL_LIBS}
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_PROGRAM_OPTIONS_LIBRARY}
-    ${Boost_REGEX_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
-    ${GLOG_LIBRARIES}
-    ${FREEIMAGE_LIBRARIES}
-    ${CERES_LIBRARIES}
-    ${OPENGL_gl_LIBRARY}
-    ${OPENGL_glu_LIBRARY}
-    ${Qt5Core_LIBRARIES}
-    ${Qt5OpenGL_LIBRARIES}
-    ${Qt5Widgets_LIBRARIES}
+    ${CONAN_LIBS}			
 )
 
 if(OPENMP_FOUND)
@@ -305,7 +302,6 @@ if(UNIX)
 endif()
 
 set(COLMAP_INTERNAL_LIBRARIES
-    flann
     graclus
     lsd
     pba
@@ -331,7 +327,6 @@ add_subdirectory(src)
 # Generate source groups for Visual Studio, XCode, etc.
 ################################################################################
 
-COLMAP_ADD_SOURCE_DIR(lib/FLANN LIB_FLANN_SRCS *.h *.cpp *.hpp *.cu)
 COLMAP_ADD_SOURCE_DIR(lib/Graclus LIB_GRACLUS_SRCS *.h *.c)
 COLMAP_ADD_SOURCE_DIR(lib/LSD LIB_LSD_SRCS *.h *.c)
 COLMAP_ADD_SOURCE_DIR(lib/PBA LIB_PBA_SRCS *.h *.cpp *.cu)
@@ -358,7 +353,6 @@ COLMAP_ADD_SOURCE_DIR(src/util UTIL_SRCS *.h *.cc)
 # intellisense can't find any of the included files).
 add_library(
     ${COLMAP_SRC_ROOT_FOLDER}
-    ${LIB_FLANN_SRCS}
     ${LIB_GRACLUS_SRCS}
     ${LIB_LSD_SRCS}
     ${LIB_PBA_SRCS}
